### PR TITLE
Fix table inserter height and width

### PR DESCRIPTION
**Pull request by juliaroldi**
Change table column and row inserter behavior to be visible only when cells at the top, left when LTR active and right when RTL active are hovered. If the middle cells are hovered or the extremity cells are merged, the inserter will not be shown. 

TableInserter

### CODE REVIEW

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts [hunk 1]
```diff
@@ -116,6 +116,7 @@ export default class TableEditor {
             let j = 0;
             for (; j < tr.cells.length; j++) {
                 const td = tr.cells[j];
+                const table = normalizeRect(this.table.getBoundingClientRect());
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts:119]: need null check, like like 122

**Comment by JiuqingSong** [comment]
table inserter 2

Still see this issue

**Comment by juliaroldi** [comment]
> table inserter 2 [ table inserter 2 ](https://user-images.githubusercontent.com/23065085/186773031-c35922d3-8789-45f4-810d-969c83284aff.gif) [ ](https://user-images.githubusercontent.com/23065085/186773031-c35922d3-8789-45f4-810d-969c83284aff.gif)
> 
> Still see this issue

I had to check if the cell above is not merged to fix this issue. 

tableInserterFix

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts [hunk 2]
```diff
@@ -116,17 +116,22 @@ export default class TableEditor {
...
                 const lessThanBottom = y <= tdRect.bottom;
                 const lessThanRight = this.isRTL ? x >= tdRect.right : x <= tdRect.right;
 
                 if (lessThanRight && lessThanBottom) {
-                    if (i == 0 && y <= tdRect.top + INSERTER_HOVER_OFFSET) {
+                    const isOnTableTop = i === 0;
+                    const isOnLeftOrRight = this.isRTL
+                        ? tdRect.right >= tableRect.right && tdRect.right <= tableRect.right + 1
+                        : tdRect.left >= tableRect.left && tdRect.left <= tableRect.left + 1;
+                    if (i == 0 && y <= tdRect.top + INSERTER_HOVER_OFFSET && isOnTableTop) {
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts:134]: i==0 and isOnTableTop are always the same, right? So we just need to check one of them

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts [hunk 3]
```diff
@@ -116,17 +116,22 @@ export default class TableEditor {
...
                     continue;
                 }
 
                 const lessThanBottom = y <= tdRect.bottom;
                 const lessThanRight = this.isRTL ? x >= tdRect.right : x <= tdRect.right;
 
                 if (lessThanRight && lessThanBottom) {
-                    if (i == 0 && y <= tdRect.top + INSERTER_HOVER_OFFSET) {
+                    const isOnTableTop = i === 0;
+                    const isOnLeftOrRight = this.isRTL
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts:131]: what is this for?

Normally when check rtl case, the condition is in opposite way of ltr case. e.g. if we check "greater than" for LTR:
```
if (a > b) {....}
```

then we check "less than" in rtl:
```if (a < b) {...}```

Not sure why here it checks the same direction in both case. Is this a typo?

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts [hunk 4]
```diff
@@ -137,10 +142,18 @@ export default class TableEditor {
                         j == 0 &&
                         (this.isRTL
                             ? x >= tdRect.right - INSERTER_HOVER_OFFSET
-                            : x <= tdRect.left + INSERTER_HOVER_OFFSET)
+                            : x <= tdRect.left + INSERTER_HOVER_OFFSET) &&
+                        isOnLeftOrRight
                     ) {
+                        const tdAbove = this.table.rows[i - 1]?.cells[0];
+                        const tdAboveRect = normalizeRect(tdAbove.getBoundingClientRect());
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts:149]: tdAbove can be null?

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts [hunk 5]
```diff
@@ -137,12 +142,21 @@ export default class TableEditor {
...
+
+                        const isTdNotAboveMerged =
+                            this.isRTL && tdAboveRect
+                                ? tdAboveRect.right === tdRect.right
+                                : tdAboveRect.left === tdRect.left;
```
Comment (JiuqingSong) [inline packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts:154]: still, tdAboveRect can be null

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts (hunk 2 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts:134]: have you figured out this one?

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts (hunk 5 above)
Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts]: Added a check

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts (hunk 4 above)
Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts]: Added a check

File: packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts (hunk 2 above)
Comment (juliaroldi) [reply packages/roosterjs-editor-plugins/lib/plugins/TableResize/editors/TableEditor.ts]: Yes, just changed a = to be i===0

**Review by JiuqingSong — APPROVED** [review]

**Pull request merged**